opencog/nlp/sureal/SuRealPMCB.cc

```diff
@@ -99,20 +99,123 @@ bool SuRealPMCB::variable_match(const Handle &hPat, const Handle &hSoln)
     if (hSolnWordInst == Handle::UNDEFINED)
         return false;
 
-    // get the source connectors for the solution
-    std::string scmCode = "(ListLink (word-inst-get-source-conn " + SchemeSmob::to_string(hSolnWordInst) + "))";
-    HandleSeq qTargetConns = m_as->get_outgoing(m_eval->eval_h(scmCode));
+    // the source connectors for the solution
+    HandleSeq qTargetConns;
 
-    HandleSeq qOr = get_neighbors(hPatWordNode, false, true, LG_WORD_CSET, false);
-    HandleSeq qDisjuncts;
+    // get all the ListLinks connecting to hSolnWordInst
+    HandleSeq qSolnListLinks;
+    hSolnWordInst->getIncomingSetByType(back_inserter(qSolnListLinks), LIST_LINK);
+
+    // helper for checking if a ListLink has a LgLinkInstanceNode as a neighbor
+    auto hasNoLgLinkInstanceNode = [](Handle& h)
+    {
+        HandleSeq qN = get_neighbors(h, true, false, EVALUATION_LINK);
+        return not std::any_of(qN.begin(), qN.end(), [](Handle& hn) { return hn->getType() == LG_LINK_INSTANCE_NODE; });
+    };
+
+    // remove any ListLink that doesn't have a LgLinkInstanceNode as a neighbor
+    qSolnListLinks.erase(std::remove_if(qSolnListLinks.begin(), qSolnListLinks.end(), hasNoLgLinkInstanceNode), qSolnListLinks.end());
```

Comment:
Is this doing the same as https://github.com/opencog/atomspace/blob/master/opencog/atomutils/AtomUtils.h#L131